src/lib/operators/join_hash.cpp

```diff
@@ -129,16 +129,17 @@ std::vector<std::shared_ptr<HashTable<HashedType>>> build(const RadixContainer<L
 
   for (size_t current_partition_id = 0; current_partition_id < (radix_container.partition_offsets.size() - 1);
        ++current_partition_id) {
-    jobs.emplace_back(std::make_shared<JobTask>([&, current_partition_id]() {
-      auto& partition_left = static_cast<Partition<LeftType>&>(*radix_container.elements);
-      const auto& partition_left_begin = radix_container.partition_offsets[current_partition_id];
-      const auto& partition_left_end = radix_container.partition_offsets[current_partition_id + 1];
-      const auto partition_size = partition_left_end - partition_left_begin;
+    const auto& partition_left_begin = radix_container.partition_offsets[current_partition_id];
+    const auto& partition_left_end = radix_container.partition_offsets[current_partition_id + 1];
+    const auto partition_size = partition_left_end - partition_left_begin;
 
-      // Prune empty partitions, so that we don't have too many empty hash tables
-      if (partition_size == 0) {
-        return;
-      }
+    // Prune empty partitions, so that we don't have too many empty hash tables
+    if (partition_size == 0) {
+      continue;
+    }
+
+    jobs.emplace_back(std::make_shared<JobTask>([&, current_partition_id, partition_size]() {
```

Comment:
This looks like you are capturing the extracted variables by reference. By the time the lambda is run, things like `partition_left_begin` might have already changed. Have you tested this in multi-threaded mode?

Reply:
I have tested it in multi-threaded mode. The things haven't changed when the lambda was run.
However, passing the [u]ints by reference doesn't make sense.

Reply:
Well no - you *should not* pass them by reference, but by copy. By adding a sleep and then printing `partition_left_begin`, you should see that they are all referring to the same value.

Reply:
Okay. I am passing the variables by copy now.